client/state/themes/selectors/get-theme-signup-url.js

```diff
@@ -57,12 +58,19 @@ function getThemeTypeUrlParameter( state, themeId ) {
 		return 'premium';
 	}
 
+	if ( isWpcomTheme( state, themeId ) && ! isThemePremium( state, themeId ) ) {
```

Comment:
I don't think you need to check `isThemePremium` since you already checked it in line 57.

Reply:
Yeah, it's redundant. But I preferred it like this because the condition is more robust, especially for future additions (if we need it). :)